StripeFinancialConnections/StripeFinancialConnections/Source/Common/TestModeAutofillBannerView.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  TestModeAutofillBannerView.swift
+//  StripeFinancialConnections
+//
+//  Created by Mat Schmid on 2024-06-20.
+//
+
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+import UIKit
+
+class TestModeAutofillBannerView: UIView {
+
+    enum Context {
+        case account
+        case otp
+
+        var buttonLabel: String {
+            switch self {
+            case .account:
+                STPLocalizedString(
+                    "Use test account",
+                    "Label for a button shown to a user who is in test mode, which will give them the option to autofill mock account credentials."
+                )
+            case .otp:
+                STPLocalizedString(
+                    "Use test code",
+                    "Label for a button shown to a user who is in test mode, which will give them the option to autofill a mock one time passcode."
+                )
+            }
+        }
+    }
+
+    private let context: Context
+    private let didTapAutofill: () -> Void
+
+    // MARK: - Subviews
+
+    private lazy var messageLabel: UILabel = {
+        // Create icon as a text attachement.
+        let icon = Image.info
+            .makeImage(template: true)
+            .withTintColor(.attention300)
+        let textAttachment = NSTextAttachment(image: icon)
```

Comment:
I haven't done it this way, but if it works well it could be better (because then iOS handles the text wrapping)

The things to consider are:
- does this set the image to the right size?
- does it get us the correct spacing?
- how does it work with different text (ex. localization) and different font sizes (ex. dynamic font)? does the text wrap/scale correctly?

Reply:
I checked web and I think they just do an HStack (but, of course, we should do whatever is best for us as long as it handles everything equal/better):

<img width="336" alt="Screenshot 2024-06-21 at 10 16 00 AM" src="https://github.com/stripe/stripe-ios/assets/105514761/368e8093-4fea-488f-aa68-786f65e44c30">

Reply:
Here is the screen while using the largest dynamic font: 

<img width=50% src="https://github.com/stripe/stripe-ios/assets/172562065/a538a599-0338-486f-8d6d-6df03d92f426">